app/jobs/fax_application_job.rb

```diff
@@ -1,25 +1,18 @@
 class FaxApplicationJob < ApplicationJob
-  def perform(snap_application_id:)
-    @snap_application = SnapApplication.find(snap_application_id)
-    return if snap_application.faxed?
-    Fax.send_fax(
-      number: fax_recipient.number,
-      file: snap_application.pdf.path,
-      recipient: fax_recipient.name,
-    )
-    snap_application.update(faxed_at: Time.zone.now)
-  ensure
-    snap_application.pdf.try(:close)
-    snap_application.pdf.try(:unlink)
-  end
-
-  private
+  def perform(export_id:)
+    export = Export.find(export_id)
+    export.execute do |snap_application|
+      return if snap_application.faxed?
 
-  attr_reader :snap_application
+      fax_recipient = FaxRecipient.new(residential_address:
+                                       snap_application.residential_address)
+      Fax.send_fax(
+        number: fax_recipient.number,
+        file: snap_application.pdf.path,
+        recipient: fax_recipient.name,
+      )
 
-  def fax_recipient
-    @fax_recipient ||=
-      FaxRecipient.new(residential_address:
-                         snap_application.residential_address)
+      "Faxed to #{fax_recipient.name} (#{fax_recipient.number})"
```

Comment:
same comment re: logging

Reply:
These are what is stored as metadata after the export happens so that the admin dashboard says what actually happened. I didn't want to make the data structured just yet sicne Iw asn't sure which data would be needed.

Reply:
gotcha - any way to make the fact that it's metadata more obvious? To me these read as random strings! 🍓

Reply:
I've been thinking about this, and it may be an indication we're mixing command/query separation. Perhaps the jobs should have a "success_message" in their interface? And then assume they succeeded in the event that no exceptions were thrown?

Reply:
And I just realized that since this is using a block as an interface, we don't have a reference to the actual job... which... uhh... I feel conflicted about.